ruby/reschedule_or_cancel_unfilled_service.rb

```diff
@@ -0,0 +1,216 @@
+# Service for handling the automatic rescheduling/canceling of unfilled bookings.
+module RescheduleOrCancelUnfilledService
+  MAX_OFFSET = 8 * 7 # up to 8 weeks away
+
+  class << self
+    def auto_rbu_or_cbu_candidate?(booking, date_start: booking.date_start)
+      return false unless booking.confirmed
+      return false if booking.provider.present?
+
+      window = auto_rbu_or_cbu_minutes(region: booking.region, service: booking.service).minutes
+
+      ((date_start - window)...date_start).cover?(Time.current)
+    end
+
+    # Get bookings in need of auto-RBU/CBU
+    # @return array of eligible bookings
+    def bookings_to_reschedule_or_cancel
+      regions  = Region.select { |region| ConfigParam.regional_auto_rbu?(region) }
+      start_at = Time.current
+      window = [RegionAutoAcceptWindow.maximum(:auto_rbu_minutes).to_i, RegionAutoAcceptWindow::AUTO_RBU_WINDOW_MINUTES].max
```

Comment:
We need to refactor this so that we can understand what's doing. Also, we may want to check first if there are records in the database before doing the max.